File: app/src/main/java/com/sun/replugin/SunApplication.java
package com.sun.replugin;

import android.content.Context;
import com.qihoo360.replugin.RePluginApplication;
import com.qihoo360.replugin.RePluginConfig;

/**
 * Created by hengyangji on 2022/6/27
 */
public class SunApplication extends RePluginApplication {

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        TestApplicationLoadClass.load();
        HostServiceProvider.services.clear();
        HostServiceProvider.services.add(new HostServiceImpl());
    }
    @Override
    protected RePluginConfig createConfig() {
        RePluginConfig config = new RePluginConfig();
        config.setUseHostClassIfNotFound(true);
        return config;
    }
}

File: app/src/main/java/com/sun/replugin/SunApplication.kt
package com.sun.replugin

import android.content.Context
import android.util.Log
import com.qihoo360.replugin.RePluginApplication

/**
 * Created by hengyangji on 2022/6/20
 */
class SunApplication : RePluginApplication() {
    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
        Log.i("SunApplication", "attachBaseContext")
    }

    override fun onCreate() {
        super.onCreate()
        Log.i("SunApplication", "onCreate")
    }
}